cmd/controller/controller_test.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/arschles/assert"
+)
+
+// TODO: add more tests
+
+func Test_resolveURL(t *testing.T) {
+	baseURL := "http://charts.example.com"
```

Comment:
I think your baseURL is conveniently chosen here, and is hiding bugs.  Since `baseURL` is `repoURL` in our use case, a more realistic testcase might be: `resolveURL("https://bitnami.com/repo/index.yaml", "mychart.tgz")` -> `https://bitnami.com/repo/mychart.tgz`

Reply:
Why would the repo URL contain the index.yaml resource? That's an incorrect repo URL and Helm CRD should error if it finds that, rather than silently accept it. This is expected to fail: `helm repo add bitnami https://bitnami.com/repo/index.yaml`, so we should too.